packages/multi-test/src/app.rs

```diff
@@ -62,14 +63,15 @@ where
     C: Clone + fmt::Debug + PartialEq + JsonSchema + 'static,
 {
     pub fn new(
-        api: Box<dyn Api>,
```

Comment:
Nice to remove Box from the public API. Better this way.

(And on reflection from you, I think maybe generics for eg. Bank implementation makes sense.. but that is a different refactor).

Reply:
I would actually prefer to be generic over all those things (like `Api` and `Storage`), as internal optimizer tells me there is additional overhead of v-call here, but... come on, those are tests. This makes usage easier, as additional type hints are never an issue. I would not do that unless there is an actual need (I found sometimes dyn traits doesn't play well with typesystem, but I don't think here is a case).

Reply:
If you look at `Deps` and `DepsMut` we use `dyn` for `Storage`, `Api` and `Querier`. That change was made in 0.10 or 0.11. This made writing contracts much, much easier, especially for Rust newcomers.

I am very happy with that and would avoid re-introducing generics there. But for all other types, I am open. I just got in the habit after seeing how much that simplified the code.